lib/client/keyagent_test.go

```diff
@@ -248,26 +248,37 @@ func TestHostCertVerification(t *testing.T) {
 	// Create a CA, generate a keypair for the CA, and add it to the known
 	// hosts cache (done by "tsh login").
 	keygen := testauthority.New()
-	caPriv, caPub, err := keygen.GenerateKeyPair()
-	require.NoError(t, err)
-	caSigner, err := ssh.ParsePrivateKey(caPriv)
-	require.NoError(t, err)
-	caPublicKey, _, _, _, err := ssh.ParseAuthorizedKey(caPub)
-	require.NoError(t, err)
-	err = lka.keyStore.AddKnownHostKeys("example.com", s.hostname, []ssh.PublicKey{caPublicKey})
-	require.NoError(t, err)
+
+	generateCA := func(hostname string) (ssh.Signer, auth.TrustedCerts) {
+		caPriv, caPub, err := keygen.GenerateKeyPair()
+		require.NoError(t, err)
+		caSigner, err := ssh.ParsePrivateKey(caPriv)
+		require.NoError(t, err)
+		caPublicKey, _, _, _, err := ssh.ParseAuthorizedKey(caPub)
+		require.NoError(t, err)
+		err = lka.keyStore.AddKnownHostKeys(hostname, s.hostname, []ssh.PublicKey{caPublicKey})
+		require.NoError(t, err)
+
+		_, trustedCerts, err := newSelfSignedCA(caPriv, hostname)
+		require.NoError(t, err)
+		trustedCerts.ClusterName = hostname
+		return caSigner, trustedCerts
+	}
+
+	rootCASigner, rootTrustedCerts := generateCA("example.com")
```

Comment:
This code is flaky. I fixed it in master - please cherry-pick the fix into this PR as well.